files/common/var/lib/delphix-platform/clear-sharenfs

```diff
@@ -0,0 +1,48 @@
+#!/usr/bin/python3
+
+import subprocess
+
+lines = subprocess.check_output(
+    "zfs list -t filesystem -r -H -o name,sharenfs,mounted domain0",
+    shell=True).decode('utf-8').splitlines()
+datasets = [line.split('\t') for line in lines]
+# Filter out datasets with sharenfs == "off" and sharenfs == "-"
+datasets = [ds for ds in datasets if ds[1] not in ('off', '-')]
+
+# We only clear sharenfs for datasets that are mounted as in some cases
+# zfs can mount datasets while clearing their sharenfs property and mount
+# children without mounting the parent datasets first. Given that ZFS
+# automatically creates directories for mountpoints, mounting children
+# before the parents would end up creating sub-directories under their
+# parent's mountpoints and so parents would fail to mount later. This
+# is something we really want to avoid, so as a precaution we make sure
+# that all the datasets that we manipulate are already mounted.
+unmounted_datasets = [ds for ds in datasets if ds[2] != 'yes']
+datasets = [ds for ds in datasets if ds not in unmounted_datasets]
+
+if unmounted_datasets:
```

Comment:
What are the ramifications of skipping these datasets as it pertains to migration?

Reply:
I haven't tested what exact ramifications would there be, I'd guess there might be issues starting those datasets. I could try to test this out by sharing some datasets on Illumos, unmounting them and then going through the migration.

That said, I'm not sure if there are any cases where we would have unmounted datasets with sharenfs set, so I've just added this code as a failsafe.